From 3826296f43c3ea96648012f4ef5946f8b78f9a5a Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Mon, 22 Dec 2025 14:17:10 -0500
Subject: [PATCH 430/473] winewayland: Unset fullscreen before setting
 different output hint.

Seems like this is required.
---
 dlls/winewayland.drv/wayland_surface.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/winewayland.drv/wayland_surface.c b/dlls/winewayland.drv/wayland_surface.c
index 4c6181e09f9..90aec786839 100644
--- a/dlls/winewayland.drv/wayland_surface.c
+++ b/dlls/winewayland.drv/wayland_surface.c
@@ -698,6 +698,8 @@ static void wayland_surface_reconfigure_geometry(struct wayland_surface *surface
             {
                 TRACE("Updating fullscreen output: output %p, old output %p\n",
                     output, surface->requested_output);
+                xdg_toplevel_unset_fullscreen(surface->xdg_toplevel);
+                wl_display_flush(process_wayland.wl_display);
                 xdg_toplevel_set_fullscreen(surface->xdg_toplevel, output);
                 wl_display_flush(process_wayland.wl_display);
                 surface->requested_output = output;
-- 
2.52.0

